graphix/gflow.py

```diff
@@ -461,3 +388,356 @@ def get_layers(l_k):
     for i in l_k.keys():
         layers[l_k[i]].append(i)
     return d, layers
+
+
+class GF2Solver:
+    """Solver for GF(2) linear equations"""
```

Comment:
any reference or description of the algorithm?

Reply:
I didn't refer to anything in particular.
I generalize well-known Gauss-Jordan elimination onto GF2 and non-square matrix.

Step 1. Conduct row elimination and column pivoting when necessary(`GF2Solver.forward_elimination`). We get a standard form of n x m matrix and kernel of matrix here.
Step 2. We get solutions for each node(`GF2Solver.backward_substitution`).
Step 3. If mode is 'abstract', the solver returns a set of `SolutionNode` and uncertainty nodes(kernel of matrix). If mode is 'single' or 'all', the solver substitutes uncertainty map and returns concrete solutions.

Reply:
good, I think what you wrote above seems to be a good docstring for this class?

Reply:
I updated the docstring just now